### PR TITLE
EMP-387 fix the anticipated preparation time format

### PR DESCRIPTION
**Pull request by solomonmanik**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-XXX)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/mapper/Crm5Mapper.java [hunk 1]
```diff
@@ -125,4 +125,15 @@ public interface Crm5Mapper {
     @Mapping(target="solicitor.certification.name", source="certification_sol_name")
     @Mapping(target="officeUseOnly.decision", source="decision")
     CRM5DetailsDTO getDetailsDTOFromModel(Crm5DetailsModel crm5DetailsModel);
+
+    default String processAcPreparationTime(Crm5DetailsModel crm5DetailsModel) {
+        if (null != crm5DetailsModel.getAc_preparation_time()) {
```
Comment (tanovellino) [inline equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/mapper/Crm5Mapper.java]: this function does not need to be a default one.
Maybe we can have a generic: convertToTimeSpentString() and move it to the interface and have it as a generic there

File: equinity-historical-data/src/test/resources/Crm5MockFile_5001604.txt [hunk 2]
```diff
@@ -1821,7 +1821,7 @@
 "Originator_tl_neverpermanentlyremove": 0,
 "Ac_letters": 2,
 "Additional_text": "",
-"Ac_preparation_time": "00:15:00",
+"Ac_preparation_time": "1899-12-31T10:00:00",
```
Comment (tanovellino) [inline equinity-historical-data/src/test/resources/Crm5MockFile_5001604.txt:1824]: do we have a test for the other case?
case 1: input="00:15:00", output ="00:15:00"
case 2: input="1899-12-31T10:00:00", output="00:15:00"
case 3: input="30 minutes", output=" minutes"
case 4:  input="30 min", output="30 min"
case 5: input"100:59:00", output="00:59:00".. but it should be "100:59:00"

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/mapper/Crm5Mapper.java [hunk 3]
```diff
@@ -125,4 +125,17 @@ public interface Crm5Mapper {
     @Mapping(target="solicitor.certification.name", source="certification_sol_name")
     @Mapping(target="officeUseOnly.decision", source="decision")
     CRM5DetailsDTO getDetailsDTOFromModel(Crm5DetailsModel crm5DetailsModel);
+
+    default String processAcPreparationTime(Crm5DetailsModel crm5DetailsModel) {
```
Comment (tanovellino) [inline equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/mapper/Crm5Mapper.java:129]: this function does not need to be a default one.
Maybe we can have a generic: convertToTimeSpentString() and move it to the interface and have it as a generic there

File: equinity-historical-data/src/test/resources/Crm5MockFile_5001604.txt (hunk 2 above)
Comment (solomonmanik) [reply equinity-historical-data/src/test/resources/Crm5MockFile_5001604.txt:1824]: Case 1 & case 2 are valid, the data in the DB suggests that Case3,4 & 5 are not valid.
I'll add two test cases.

Comment (tanovellino) [reply equinity-historical-data/src/test/resources/Crm5MockFile_5001604.txt:1824]: cool. Maybe just add a backlog ticket to add the possibility of a regex check for format. But not prio now

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/mapper/Crm5Mapper.java (hunk 3 above)
Comment (solomonmanik) [reply equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/mapper/Crm5Mapper.java]: Done

File: equinity-historical-data/src/test/resources/Crm5MockFile_5001604.txt (hunk 2 above)
Comment (solomonmanik) [reply equinity-historical-data/src/test/resources/Crm5MockFile_5001604.txt]: Okay.  Tests added for case 1 & 2.

**Pull request merged**